submissions/Siusarna/data-structures/data-structures/stack.js

```diff
@@ -0,0 +1,29 @@
+/* eslint-disable no-underscore-dangle */
+
+const Node = require('./Element.js');
+
+module.exports = class Stack {
+  constructor() {
+    this._head = null;
+    this._size = 0;
+  }
+
+  push(inputValue) {
+    this._head = new Node(inputValue, this._head);
+    this._size += 1;
+  }
+
+  pop() {
+    if (this._size) {
+      const removedValue = this._head.getValue();
+      this._head = this._head.getNext() || null;
+      this._size -= 1;
+      return removedValue;
+    }
+    throw new Error('Stack is empty');
```

Comment:
Do you mean it's better to just return an empty element?

Reply:
I just don't think that removing element from empty collection should produce error. 
I mean, that it is not error by definition - everything worked well and code don't have a mistakes. 

What you mean under "empty element"? Instance of `Node`? I am not sure, this will assume, that you **had** this node in your collection, no? 🤔

Reply:
I understood you, so return string "Stack is empty" is better solution than throw error?)

Reply:
I understood you, so return string "Stack is empty" is better solution than throw error?)

Reply:
Then type of returned value will be `string` which doesn't mean "empty".
Also, to check was popping successful or not I will need to write check like:
```javascript
const wasRemoved = stack.pop() === 'Stack is empty'
```
Which is too long, as for me.
Instead, JS has `undefined`. You can use it, as well as `Array#pop` used.